### PR TITLE
Adding support for GDAL scale/offset metadata

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: plugin/tiff/src/main/java/it/geosolutions/imageioimpl/plugins/tiff/TIFFImageReader.java [hunk 1]
```diff
@@ -1265,6 +1291,30 @@ private void initializeFromMetadata() {
             }
         }
 
+        // scales and offsets, if any
+        this.offsets = null;
+        this.scales = null;
+        f = imageMetadata.getTIFFField(PrivateTIFFTagSet.TAG_GDAL_METADATA);
+        this.offsets = null;
+        this.scales = null;
```
Comment (dromagnoli) [inline plugin/tiff/src/main/java/it/geosolutions/imageioimpl/plugins/tiff/TIFFImageReader.java]: duplicate initialization?
        this.offsets = null;
        this.scales = null;

File: plugin/tiff/src/main/java/it/geosolutions/imageioimpl/plugins/tiff/gdal/GDALMetadata.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        return getBandDoubleValues(bands, "offset");
+    }
+
+    /**
+     * Utility methods extracting an offset array from the items, if available. Will throw runtime
+     * exceptions if the offset item values are not numbers.
+     *
+     * @return the offset array, or null if no items are present, or no offset was found
+     */
+    public Double[] getScales(int bands) {
```
Comment (dromagnoli) [inline plugin/tiff/src/main/java/it/geosolutions/imageioimpl/plugins/tiff/gdal/GDALMetadata.java]: copy and paste :)
"offset" should be "scale" in this javadoc

File: plugin/tiff/src/main/java/it/geosolutions/imageioimpl/plugins/tiff/TIFFImageReader.java (hunk 1 above)
Comment (aaime) [reply plugin/tiff/src/main/java/it/geosolutions/imageioimpl/plugins/tiff/TIFFImageReader.java]: Meglio abbondare!
(I'll fix it :-p )

**Comment by aaime** [comment]
Feedback applied.

**Comment by dromagnoli** [comment]
merged, 
are you planning to backport it?

**Comment by aaime** [comment]
I do, but not right away as it needs geotools changes to be working.

**Comment by dromagnoli** [comment]
sure. just remind me that before we do the 1.2.x release :)

**Pull request merged**